wled00/wled_server.cpp

```diff
@@ -482,7 +482,17 @@ void initServer()
       return;
     }
 
-    if(handleSet(request, request->url())) return;
+    if (request->url().indexOf("win") == 0) {
+      if (handleHttpApi(request->url().substring(3))) {
+        if (request->url().indexOf(F("&FX=")) > 0) {
```

Comment:
Wouldn't it be better to move this into `handleHttpApi()`?
IMO it's place is there.

Reply:
In the original code, we call `unloadPlaylist()` only the FX was changed from an HTTP request, but not from any other context.  It made more sense to me to place that code only in the HTTP request context, rather than add a special parameter to `handleHttpApi()` *just* to switch this call on or off.  It does mean naming the tag twice, but I think that's the lesser evil -- https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction

An alternative might be to augment the return code to indicate if the FX value was changed, rather than parsing the tag multiple times.

Reply:
I agree -- although this PR was aimed to be a refactor with no behavioural changes, and the original logic only unloaded the playlist in the context of an HTTP request.

I'll do a more thorough review of the other call contexts and see if I can draw a better map.  No reason not to further improve things as we go.